### PR TITLE
keepalived service: init

### PR DESCRIPTION
**Pull request by mbrgm**

### CODE REVIEW

File: nixos/modules/services/networking/keepalived.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.keepalived = {
+      description = "Keepalive Daemon (LVS and VRRP)";
+      after = [ "network.target" "syslog.target" ];
+      wants = [ "network.target" ];
```
Comment (joachifm) [inline nixos/modules/services/networking/keepalived.nix]: Explicit dependency on `network.target` is probably redundant, it'll be part of normal bootup anyway. Also, `network.target` is very weakly defined. All you can be sure of is that your service will be stopped *before* network on shutdown ... If you intended to require network connectivity or something, you'd depend on and order after `network-online.target`, that at least is semantically more correct.

File: nixos/modules/services/networking/keepalived.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Keepalived.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
```
Comment (joachifm) [inline nixos/modules/services/networking/keepalived.nix]: I think this default value will fail to evaluate (unless explicitly converted to a string with `toString`). I think just using an empty string here is okay anyway.

**Comment by mbrgm** [comment]
Note that build breakage did not originate from this change.

**Comment by mbrgm** [comment]
@fpletz From my side, you can remove the WIP label -- the last commit is the final implementation.

File: nixos/modules/services/networking/keepalived/default.nix [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Whether to enable the builtin AgentX subagent.
+          '';
+        };
+
+        socket = mkOption {
+          type = types.nullOr types.string;
```
Comment (joachifm) [inline nixos/modules/services/networking/keepalived/default.nix]: I suspect you want to use `types.str`

File: nixos/modules/services/networking/keepalived/vrrp-options.nix [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      default = "PASS";
+      description = ''
+        Authentication type:
+
+        PASS - Simple password (suggested)
+        AH - IPSEC (not recommended))
+      '';
+    };
+
+    authPass = mkOption {
```
Comment (joachifm) [inline nixos/modules/services/networking/keepalived/vrrp-options.nix]: Looks like this ends up in the Store (and so will be world-readable), is that a problem?

File: nixos/modules/services/networking/keepalived/default.nix (hunk 3 above)
Comment (mbrgm) [inline nixos/modules/services/networking/keepalived/default.nix]: Yup, you're right!

File: nixos/modules/services/networking/keepalived/vrrp-options.nix (hunk 4 above)
Comment (mbrgm) [inline nixos/modules/services/networking/keepalived/vrrp-options.nix]: I looked this up once more and found that authentication was deprecated in [RFC3768](https://www.ietf.org/rfc/rfc3768.txt) and is only there for backwards compatibility. I'm going to remove the whole authentication section so people don't use a deprecated mechanism.

**Comment by mbrgm** [comment]
@joachifm I added two commits addressing the issues you mentioned. Does this match what you had in mind?

**Comment by joachifm** [comment]
Thank you

**Pull request merged**